### PR TITLE
lsd: 0.23.1 -> 1.0.0, modernise packages, thin out maintainers

### PR DESCRIPTION
**Pull request by SuperSandro2000**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/lsd/default.nix [hunk 1]
```diff
@@ -9,23 +9,26 @@
 
 rustPlatform.buildRustPackage rec {
   pname = "lsd";
-  version = "0.23.1";
+  version = "1.0.0";
 
   src = fetchFromGitHub {
     owner = "Peltoche";
```
Comment (marsam) [inline pkgs/tools/misc/lsd/default.nix:15]:
```suggestion
    owner = "lsd-rs";
```

File: pkgs/tools/misc/lsd/default.nix [hunk 2]
```diff
@@ -9,23 +9,26 @@
...
-    repo = pname;
-    rev = version;
-    sha256 = "sha256-FY1odcKBl7zJ+MxfohkmC1e45fPQK3MKB3orQdCRpA4=";
+    repo = "lsd";
+    rev = "v${version}";
+    hash = "sha256-FY1odcKBl7zJ+MxfohkmC1e45fPQK3MKB3orQdCRpA4=";
```
Comment (marsam) [inline pkgs/tools/misc/lsd/default.nix:18]: The hash needs to be updated

File: pkgs/tools/misc/lsd/default.nix [hunk 3]
```diff
@@ -39,6 +42,6 @@ rustPlatform.buildRustPackage rec {
     homepage = "https://github.com/Peltoche/lsd";
```
Comment (marsam) [inline pkgs/tools/misc/lsd/default.nix:42]:
```suggestion
    homepage = "https://github.com/lsd-rs/lsd";
```

**Review by marsam — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/lsd/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/tools/misc/lsd/default.nix:18]: whopsie 😓

File: pkgs/tools/misc/lsd/default.nix [hunk 4]
```diff
@@ -9,23 +9,26 @@
...
-    rev = version;
-    sha256 = "sha256-FY1odcKBl7zJ+MxfohkmC1e45fPQK3MKB3orQdCRpA4=";
+    owner = "lsd-rs";
+    repo = "lsd";
+    rev = "v${version}";
+    hash = "sha256-syT+1LNdigUWkfJ/wkbY/kny2uW6qfpl7KmW1FjZKR8=";
   };
 
-  cargoSha256 = "sha256-t7J7hIbLlRq99Yd2/3Zn+PbHhJtaJRdDluDXN0Hp/Jc=";
+  cargoHash = "sha256-viLr76Bq9OkPMp+BoprQusMDgx59nbevVi4uxjZ+eZg=";
```
Comment (marsam) [inline pkgs/tools/misc/lsd/default.nix:22]: Would you mind adding the following?
```nix
buildInputs = lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.Security ];
```
Besides that, LGTM

**Review by marsam — APPROVED** [review]

**Pull request merged**